mongoengine/connections_manager.py

```diff
@@ -0,0 +1,110 @@
+from collections import defaultdict
+
+from mongoengine.connection import get_connection, get_db
+
+__all__ = ['InvalidCollectionError', 'connection_manager']
+
+
+class InvalidCollectionError(Exception):
+    pass
+
+
+class ConnectionManager(object):
+    connections_registry = defaultdict(dict)
+
+    def get_and_setup(self, doc_cls, alias=None, collection_name=None):
+        if alias is None:
+            alias = doc_cls._get_db_alias()
+
+        if collection_name is None:
+            collection_name = doc_cls._get_collection_name()
+
+        _collection = self.connections_registry[alias].get(collection_name)
+        if not _collection:
+            _collection = self.get_collection(doc_cls, alias, collection_name)
+            if doc_cls._meta.get('auto_create_index', True):
+                doc_cls.ensure_indexes(_collection)
+            self.connections_registry[alias][collection_name] = _collection
+        return self.connections_registry[alias][collection_name]
+
+    @classmethod
+    def _get_db(cls, alias):
+        """Some Model using other db_alias"""
```

Comment:
The docstring here is pretty confusing. What is "Some Model"? Why is it using "other db_alias"? Why do we need a private method that only proxies a call to a public `get_db`?